sdk/core/core-http/src/serializer.ts

```diff
@@ -885,8 +885,14 @@ function deserializeCompositeType(
           propertyObjectName,
           options
         );
-      } else if (propertyMapper.xmlIsMsText && responseBody[XML_CHARKEY] !== undefined) {
-        instance[key] = responseBody[XML_CHARKEY];
+      } else if (propertyMapper.xmlIsMsText) {
+        if (responseBody[XML_CHARKEY] !== undefined) {
+          instance[key] = responseBody[XML_CHARKEY];
+        } else if (typeof responseBody === "string") {
+          // The special case where xml parser parses "<Name>content</Name>" into JSON of
```

Comment:
ah so the collapsing is a special quirk of the xml parser rather than something strange we were doing?

Reply:
Yes. I experimented with a couple of `xml2js` parser options but none of them worked well to our expectation.